include/mvdtool/bits/mvd3_misc.hpp

```diff
@@ -0,0 +1,400 @@
+/*
+ * Copyright (C) 2015 Adrien Devresse <adrien.devresse@epfl.ch>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ */
+#ifndef MVD3_MISC_HPP
+#define MVD3_MISC_HPP
+
+#include <algorithm>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <highfive/H5DataSet.hpp>
+
+#include "../mvd3.hpp"
+#include "../mvd_except.hpp"
+#include "../utils.hpp"
+
+namespace {
+
+inline bool is_empty(const MVD3::Range& range) {
```

Comment:
That has been there since some time, and I believe Adrien didnt want to change the public API. But I definitely like the idea. While at it, maybe a factory `Range::all()` would be nice!